### PR TITLE
js deduplicating

### PR DESCRIPTION
**Pull request by mazerakham**

### CODE REVIEW

File: src/bowser/node/DomNode.java [hunk 1]
```diff
@@ -222,7 +222,21 @@ public DomNode javascript(String name, boolean defer) {
...
+    if (isDuplicateJS(s)) {
+      return this;
+    } else {
+      return add(s);
+    }
+  }
+
+  private boolean isDuplicateJS(DomNode s) {
+    for (DomNode child : this.getChildren()) {
+      if ("script".equals(child.tag) && child.hasAttribute("src")
```
Comment (mirraj2) [inline src/bowser/node/DomNode.java:234]: you don't need to check hasAttribute if you use Object.equals()

File: src/bowser/node/DomNode.java [hunk 2]
```diff
@@ -222,7 +222,21 @@ public DomNode javascript(String name, boolean defer) {
...
     }
-    return add(s);    
+    if (isDuplicateJS(s)) {
+      return this;
+    } else {
+      return add(s);
+    }
+  }
+
+  private boolean isDuplicateJS(DomNode s) {
```
Comment (mirraj2) [inline src/bowser/node/DomNode.java:232]: I think it a cleaner function would be something like:

public DomNode getJavascriptNode(String src)

I tend to like passing the minimal amount of information into methods b/c it makes it clearer what the input is.

File: src/bowser/node/DomParser.java [hunk 3]
```diff
@@ -97,7 +97,8 @@ private void parse(Head head, DomNode parent, String s, int start, int end) {
           || node.tag.equalsIgnoreCase("code") || node.tag.equalsIgnoreCase("svg")) {
         node.add(new TextNode(s.substring(endTag + 1, endTagIndex)));
       } else {
-        parse(head, node, s, endTag + 1, endTagIndex); // add a child
+        parse(head, node, s, endTag + 1, endTagIndex); // Recursive step: parse the children of a node, adding them to
```
Comment (mirraj2) [inline src/bowser/node/DomParser.java:100]: if your comment is going to wrap, I think it's cleaner to just put it on the line above, and then it won't wrap

File: src/bowser/node/DomParser.java [hunk 4]
```diff
@@ -68,7 +68,7 @@ private void parse(Head head, DomNode parent, String s, int start, int end) {
     int endTag = s.indexOf('>', start);
 
     String tagData = s.substring(start + 1, endTag);
-    List<String> m = split(tagData, ' ');
+    List<String> m = split(tagData, ' '); // Note: this split escapes " characters.
```
Comment (mirraj2) [inline src/bowser/node/DomParser.java:71]: much better to document this on the method declaration

File: src/bowser/node/DomNode.java (hunk 1 above)
Comment (mazerakham) [reply src/bowser/node/DomNode.java:234]: Done.

File: src/bowser/node/DomNode.java (hunk 2 above)
Comment (mazerakham) [reply src/bowser/node/DomNode.java:232]: Done.

File: src/bowser/node/DomParser.java (hunk 3 above)
Comment (mazerakham) [reply src/bowser/node/DomParser.java:100]: Done.

File: src/bowser/node/DomParser.java (hunk 4 above)
Comment (mazerakham) [reply src/bowser/node/DomParser.java:71]: Done.

**Pull request merged**